src/Microsoft.DotNet.Wpf/src/WpfGfx/core/hw/hwbitmapcolorsource.cpp

```diff
@@ -1421,7 +1421,20 @@ CHwBitmapColorSource::ComputeRealizationParameters(
             &uLevels
             );
 
-        Assert(pDevice->GetMinimalTextureDesc(
+        if(RenderOptions::IsGraphicHWAccelerationForRDPEnabled())
```

Comment:
How are the two branches different? (i.e which success HR result do you expect that should fail without RDP enabled?)

Reply:
When the D3DFormat is X8R8G8B8, we cannot obtain the HResult as S_OK. Therefore, we must use the SUCCEEDED macro to determine whether the result is a pass or fail by passing the HResult to it.

We are using an if-else block to prevent users who do not wish to opt-in for HW Acceleration for RDP to follow the new code flow from doing so.

Reply:
What non S_OK result do you get? This might be worth mentioned in a comment in the code.

Reply:
Also to clarify, the risk here is that we would allow WPF applications to run in more scenarios than before? Why would we want to prevent that?

Reply:
Yes, this make sense to me to add appropriate comment about non S_OK result, I will take care of that.
And due to the lack of appropriate test coverage in this area we are not enabling it for everyone.

Reply:
As far as I understand, D3DFormat refers to the format of the adapter in which it is displaying. In my opinion, it is possible that the apps you mentioned may not be running on certain devices, but switching to the SUCCEEDED macro could potentially resolve this issue and allow the apps to run on those devices. While this improvement will certainly be beneficial, but as mentioned [here](https://github.com/dotnet/wpf/pull/7684#issuecomment-1630680002) by Pankaj we can revisit this and turn on the flag for next releases.

Reply:
I don't think my comment was addressed. Whether there is a flag or not for enabling hw rendering is separate issue to this bit of code, which I would think is actually a bug. There is no working scenario that using `SUCCEEDED` can possibly break.

Reply:
@miloush Thank you for your suggestion. After thorough discussion with our internal team, we have decided to go with your suggestion and instead creating two separate branches we will be using only SUCCEEDED macro.